### DISCUSSION TITLE
Lesson 7: Starting Lottery fuction in deploy_lottery.py error

### DISCUSSION THREAD

**Discussion by brownandrew1390** [smartcontractkit/full-blockchain-solidity-course-py#477] [Q&A] [answered]
Hello everyone, wondering if I can get some assistance with this error I seem to be running into when trying to run deploy_lottery.py after creating the start_lottery function.  The error is as follows:

```VirtualMachineError: revert: Ownable: caller is not the owner
Trace step -1, program counter 1108:
  File "C:/Users/Andrew/.brownie/packages/OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/access/Ownable.sol", line 43, in Lottery.startLottery:    
         * @dev Throws if called by any account other than the owner.
         */
        modifier onlyOwner() {
            require(owner() == _msgSender(), "Ownable: caller is not the owner");
            _;
        }
```

Let me know if you have any suggestions!

**Comment by cromewar** [comment, marked as answer]
Hello @brownandrew1390 it seems you are trying to execute a function which is Ownable using a different account which does not have permissions to execute that function. So if I have to guess I would said you forgot to add the  `{"from": account}` modifier somewhere on your deploy script, if you want it please share it here so we can review it.

Best regards, Cromewar.

**Reply by brownandrew1390** [reply]
```Python
from scripts.helpful_scripts import get_account, get_contract
from brownie import Lottery, config, network

def deploy_lottery():
    account = get_account(id="freecodecamp-account")
    Lottery.deploy(
        get_contract("eth_usd_price_feed").address,
        get_contract("vrf_coordinator").address,
        get_contract("link_token").address,
        config["networks"][network.show_active()]["fee"],
        config["networks"][network.show_active()]["keyhash"],
        {"from": account},
        publish_source=config["networks"][network.show_active()].get("verify", False),
        
    )
    print("Deployed lottery!")

def start_lottery():
    account = get_account()
    lottery = Lottery[-1]
    starting_tx = lottery.startLottery({"from": account})
    starting_tx.wait(1)
    print("The lottery is started!")


def main():
    deploy_lottery()
    start_lottery()
```

**Reply by cromewar** [reply]
As you can see on the `start_lottery`function you are using `get_account`, without the `freecodecamp-account` identifier, so if freecodecamp is the account deploying the contract it means it will be the owner of that contract, and if you use another account and try to execute a `onlyOwner` function your transaction will revert.

```Python

def start_lottery():
    account = get_account()


solution:

def start_lottery():
    account = get_account(id="freecodecamp-account")
```


Best regards, Cromewar.

**Reply by brownandrew1390** [reply]
Woo! Got it to work. Thanks! Makes sense, I was a bit confused as to what was happening behind the hood but this clears it up.

**Reply by cromewar** [reply]
I'm glad it helped you, feel free to make any other questions.
